### PR TITLE
Rewrite in Go: Start, Stop, Status

### PR DESCRIPTION
**Pull request by dokshina**

### CODE REVIEW

File: cli/project/files.go [hunk 1]
```diff
@@ -47,18 +47,30 @@ func GetStateboardPidFile(projectCtx *ProjectCtx) string {
 
 func GetInstanceConsoleSock(projectCtx *ProjectCtx, instanceName string) string {
```
Comment (olegrok) [inline cli/project/files.go:48]: Do you check that its length is less than 108 symbols?

File: cli/project/project.go [hunk 2]
```diff
@@ -53,9 +56,9 @@ type ProjectCtx struct {
 	Entrypoint           string
 	StateboardEntrypoint string
 	AppDir               string
-	ConfDir              string
+	ConfPath             string
 	RunDir               string
-	WorkDir              string
+	WorkDirBase          string
```
Comment (olegrok) [inline cli/project/project.go:61]: What does "base" mean here?

File: cli/running/common.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	"strings"
+
+	"github.com/tarantool/cartridge-cli/cli/common"
+	"gopkg.in/yaml.v2"
+
+	"github.com/tarantool/cartridge-cli/cli/project"
+)
+
+const (
+	DefaultLocalRunDir   = "tmp/run"
```
Comment (olegrok) [inline cli/running/common.go:16]: Please use https://golang.org/pkg/os/#TempDir instead of `tmp`.

File: cli/pack/app_dir.go [hunk 4]
```diff
@@ -35,7 +35,16 @@ func initAppDir(appDirPath string, projectCtx *project.ProjectCtx) error {
 				panic(err)
 			}
 
-			return relPath == ".rocks" || strings.HasPrefix(relPath, ".rocks/")
+			if relPath == ".rocks" || strings.HasPrefix(relPath, ".rocks/") {
+				return true
+			}
+
+			if _, err := os.Open(src); err != nil {
+				log.Warnf("Failed to copy: %s", err)
```
Comment (olegrok) [inline cli/pack/app_dir.go:43]: `Failed to copy` what?

File: cli/running/common.go [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+			process := NewInstanceProcess(projectCtx, instance)
+			processes.Add(process)
+		}
+	}
+
+	return &processes, nil
+}
+
+func formatEnv(key, value string) string {
+	return fmt.Sprintf("%s=%s", key, value)
```
Comment (olegrok) [inline cli/running/common.go:143]: I don't sure that it's really needed but such approach doesn't support several words values?
E.g. `KEY=VALUE1 VALUE2` is in fact `KEY=VALUE1`, should be `KEY="VALUE1 VALUE2"`

File: cli/running/process.go [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+	var err error
+
+	ctx := context.Background()
+	process.cmd = exec.CommandContext(ctx, "tarantool", process.entrypoint)
+
+	process.cmd.Env = append(os.Environ(), process.env...)
+	process.cmd.Dir = process.workDir
+
+	// create run dir
+	if err := os.MkdirAll(process.runDir, 0755); err != nil {
```
Comment (olegrok) [inline cli/running/process.go:135]: Is executable bit really needed here? Why not `0644`?

File: cli/running/writer.go [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+		// prefix
+		if _, err := os.Stdout.Write([]byte(w.prefix)); err != nil {
+			return n, err
+		}
+
+		// colorize line by log level
+		matches := logLineRgx.FindStringSubmatch(string(lineBytes))
+		if matches != nil {
+			logLevel := matches[1]
```
Comment (olegrok) [inline cli/running/writer.go:64]: and `len(matches) > 0`?

File: cli/pack/app_dir.go (hunk 4 above)
Comment (dokshina) [reply cli/pack/app_dir.go:43]: All `os` errors contain filepath, so it will be printed.

File: cli/project/files.go (hunk 1 above)
Comment (dokshina) [reply cli/project/files.go]: In the old cli we checked this only for notify socket that is used directly by `cartridge-cli`. For console sock this check will be performed by `cartridge`. It looks like a good approach.

File: cli/running/common.go (hunk 5 above)
Comment (dokshina) [reply cli/running/common.go:143]: I've just checked it - everything is OK with passing multi-words values. Using `KEY="VALUE1 VALUE2"` leads to setting `KEY` equal to `'"VALUE1 VALUE2"'` (with quotes).

Comment (olegrok) [reply cli/running/common.go:143]: Well, could you add a test case for it?

Comment (dokshina) [reply cli/running/common.go:143]: Decided that we shouldn't test `os/exec`, but if someone add quotes here, tests will fail.

File: cli/commands/stop.go [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+	stopCmd.Flags().StringVar(&projectCtx.RunDir, "run-dir", "", runDirFlagDoc)
+	stopCmd.Flags().StringVar(&projectCtx.ConfPath, "cfg", "", cfgFlagDoc)
+
+	stopCmd.Flags().BoolVar(&projectCtx.WithStateboard, "stateboard", false, stateboardFlagDoc)
+	stopCmd.Flags().BoolVar(&projectCtx.StateboardOnly, "stateboard-only", false, stateboardOnlyFlagDoc)
+}
+
+var stopCmd = &cobra.Command{
+	Use: "stop [INSTANCE_NAME...] ",
```
Comment (olegrok) [inline cli/commands/stop.go:22]: Is a space between `]` and `"` really needed.

**Pull request merged**